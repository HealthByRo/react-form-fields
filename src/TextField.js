/* @flow */

import React from 'react';
import BaseField from './BaseField';
import { TextFieldElement } from './elements';
import type { FieldProps } from './types';

export default class TextField extends BaseField {
  throwMissingNameError() {
    // TODO create file with errors, add MissingParamError
    throw new Error('name param is required for TextField');
  }

  renderFieldElement(props: FieldProps) {
    return (
      <TextFieldElement {...props} />
    );
  }

  render() {
    return super.renderWithField(this.renderFieldElement);
  }
}
