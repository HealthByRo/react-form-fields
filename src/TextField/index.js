/* @flow */

import TextInput from '../TextInput';
import createFormField from '../createFormField';

// $FlowFixMe
export default createFormField(TextInput);
