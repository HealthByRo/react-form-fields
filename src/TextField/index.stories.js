import React from 'react';
import { storiesOf } from '@kadira/storybook';
import TextField from './';

storiesOf('Fields: TextField', module)
  .add('input', () => {
    const props = {
      name: 'TEXT_FIELD_NAME',
    };

    return (
      <TextField {...props} />
    );
  })
  .add('input + placeholder', () => {
    const props = {
      name: 'TEXT_FIELD_NAME',
      placeholder: 'Sample text field with placeholder',
    };

    return (
      <TextField {...props} />
    );
  })
  .add('input + placeholder + error', () => {
    const props = {
      errors: ['Sample error for field'],
      name: 'TEXT_FIELD_NAME',
      placeholder: 'Sample text field with placeholder',
    };

    return (
      <TextField {...props} />
    );
  })
  .add('input + placeholder + multiple errors', () => {
    const props = {
      errors: [
        'First error for field',
        'Second error for field',
      ],
      name: 'TEXT_FIELD_NAME',
      placeholder: 'Sample text field with placeholder',
    };

    return (
      <TextField {...props} />
    );
  })
  .add('input + label', () => {
    const props = {
      name: 'TEXT_FIELD_NAME',
      label: 'Sample label',
    };

    return (
      <TextField {...props} />
    );
  })
  .add('input + label + error', () => {
    const props = {
      errors: ['Sample error for field'],
      name: 'TEXT_FIELD_NAME',
      label: 'Sample label',
    };

    return (
      <TextField {...props} />
    );
  })
  .add('input + label + multiple errors', () => {
    const props = {
      errors: [
        'First error for field',
        'Second error for field',
      ],
      name: 'TEXT_FIELD_NAME',
      label: 'Sample label',
    };

    return (
      <TextField {...props} />
    );
  })
  .add('input + label + placeholder', () => {
    const props = {
      name: 'TEXT_FIELD_NAME',
      label: 'Sample label',
      placeholder: 'Sample text field with placeholder',
    };

    return (
      <TextField {...props} />
    );
  })
  .add('input + label + placeholder + error', () => {
    const props = {
      errors: ['Sample error for field'],
      name: 'TEXT_FIELD_NAME',
      label: 'Sample label',
      placeholder: 'Sample text field with placeholder',
    };

    return (
      <TextField {...props} />
    );
  })
  .add('input + label + placeholder + mutliple errors', () => {
    const props = {
      errors: [
        'First error for field',
        'Second error for field',
      ],
      name: 'TEXT_FIELD_NAME',
      label: 'Sample label',
      placeholder: 'Sample text field with placeholder',
    };

    return (
      <TextField {...props} />
    );
  });
