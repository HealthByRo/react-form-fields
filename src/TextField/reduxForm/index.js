/* @flow */

import TextField from '../';
import createReduxFormField from '../../createReduxFormField';

export default createReduxFormField(TextField);
