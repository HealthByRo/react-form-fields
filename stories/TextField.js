import React from 'react';
import { storiesOf } from '@kadira/storybook';
import TextField from '../src/TextField';

storiesOf('Fields.TextField', module)
  .add('input', () => {
    const props = {
      inputId: 'ID-I',
    };

    return (
      <TextField {...props} />
    );
  })
  .add('input + placeholder', () => {
    const props = {
      inputId: 'ID-IP',
      placeholder: 'Sample input field with placeholder',
    };

    return (
      <TextField {...props} />
    );
  })
  .add('input + placeholder + error', () => {
    const props = {
      errors: ['Sample error for field'],
      inputId: 'ID-IPE',
      placeholder: 'Sample input field with placeholder',
    };

    return (
      <TextField {...props} />
    );
  })
  .add('input + placeholder + multiple errors', () => {
    const props = {
      errors: [
        'First error for field',
        'Second error for field',
      ],
      inputId: 'ID-IPES',
      placeholder: 'Sample input field with placeholder',
    };

    return (
      <TextField {...props} />
    );
  })
  .add('input + label', () => {
    const props = {
      inputId: 'ID-IL',
      label: 'Sample label',
    };

    return (
      <TextField {...props} />
    );
  })
  .add('input + label + error', () => {
    const props = {
      errors: ['Sample error for field'],
      inputId: 'ID-ILE',
      label: 'Sample label',
    };

    return (
      <TextField {...props} />
    );
  })
  .add('input + label + multiple errors', () => {
    const props = {
      errors: [
        'First error for field',
        'Second error for field',
      ],
      inputId: 'ID-ILES',
      label: 'Sample label',
    };

    return (
      <TextField {...props} />
    );
  })
  .add('input + label + placeholder', () => {
    const props = {
      inputId: 'ID-ILP',
      label: 'Sample label',
      placeholder: 'Sample input field with placeholder',
    };

    return (
      <TextField {...props} />
    );
  })
  .add('input + label + placeholder + error', () => {
    const props = {
      errors: ['Sample error for field'],
      inputId: 'ID-ILPE',
      label: 'Sample label',
      placeholder: 'Sample input field with placeholder',
    };

    return (
      <TextField {...props} />
    );
  })
  .add('input + label + placeholder + mutliple errors', () => {
    const props = {
      errors: [
        'First error for field',
        'Second error for field',
      ],
      inputId: 'ID-ILPES',
      label: 'Sample label',
      placeholder: 'Sample input field with placeholder',
    };

    return (
      <TextField {...props} />
    );
  });
