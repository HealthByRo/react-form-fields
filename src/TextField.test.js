/* @flow */

import React from 'react';
import renderer from 'react-test-renderer';
import TextField from './TextField';
import type { FieldProps } from './types';

describe('<TextField />', () => {
  let tree;
  const placeholder = 'INPUT_PLACEHOLDER';
  const label = 'INPUT_LABEL';
  const renderWithProps = (props: FieldProps) => {
    tree = renderer.create(
      <TextField {...props} />
    ).toJSON();
  };
  const getPropsWithName = (props?: Object, name?: string = 'INPUT_NAME'): FieldProps => {
    const propsWithName = { ...props };

    if (!propsWithName.name) {
      propsWithName.name = name;
    }

    return propsWithName;
  };

  describe('no params:', () => {
    it('should throw error if no name passed', () => {
      // $FlowExpectedError
      expect(() => renderWithProps()).toThrowError('name param is required for TextField');
    });
  });

  describe('param name:', () => {
    it('should NOT throw any errors if name is passed', () => {
      expect(() => <TextField name="TEST_NAME" />).not.toThrowError();
    });

    it('should render only wrapped input field with defaults', () => {
      renderWithProps(getPropsWithName());

      expect(tree).toMatchSnapshot();
    });
  });

  describe('multiple params', () => {
    it('name + placeholder', () => {
      renderWithProps(getPropsWithName({ placeholder }));

      expect(tree).toMatchSnapshot();
    });

    it('name + placeholder + errors[1]', () => {
      renderWithProps(getPropsWithName({ placeholder, errors: ['SINGLE_ERROR'] }));

      expect(tree).toMatchSnapshot();
    });

    it('name + placeholder + errors[2]', () => {
      renderWithProps(getPropsWithName({ placeholder, errors: ['FIRST_ERROR', 'SECOND_ERROR'] }));

      expect(tree).toMatchSnapshot();
    });

    it('name + label', () => {
      renderWithProps(getPropsWithName({ label }));

      expect(tree).toMatchSnapshot();
    });

    it('name + label + errors[1]', () => {
      renderWithProps(getPropsWithName({ label, errors: ['SINGLE_ERROR'] }));

      expect(tree).toMatchSnapshot();
    });

    it('name + label + errors[2]', () => {
      renderWithProps(getPropsWithName({ label, errors: ['FIRST_ERROR', 'SECOND_ERROR'] }));

      expect(tree).toMatchSnapshot();
    });

    it('name + label + placeholder', () => {
      renderWithProps(getPropsWithName({ label, placeholder }));

      expect(tree).toMatchSnapshot();
    });

    it('name + label + placeholder + errors[1]', () => {
      renderWithProps(getPropsWithName({ label, placeholder, errors: ['SINGLE_ERROR'] }));

      expect(tree).toMatchSnapshot();
    });

    it('name + label + placeholder + errors[2]', () => {
      renderWithProps(getPropsWithName({ label, placeholder, errors: ['FIRST_ERROR', 'SECOND_ERROR'] }));

      expect(tree).toMatchSnapshot();
    });
  });
});
