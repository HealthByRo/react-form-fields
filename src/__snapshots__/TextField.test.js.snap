// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> multiple params name + label + errors[1] 1`] = `
<div
  className="form-item"
>
  <label
    htmlFor="TF-INPUT_NAME"
  >
    INPUT_LABEL
  </label>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    type="text"
  />
  <div
    className="msg msg--error"
  >
    SINGLE_ERROR
  </div>
</div>
`;

exports[`<TextField /> multiple params name + label + errors[2] 1`] = `
<div
  className="form-item"
>
  <label
    htmlFor="TF-INPUT_NAME"
  >
    INPUT_LABEL
  </label>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    type="text"
  />
  <div
    className="errors-list"
  >
    <div
      className="msg msg--error"
    >
      FIRST_ERROR
    </div>
    <div
      className="msg msg--error"
    >
      SECOND_ERROR
    </div>
  </div>
</div>
`;

exports[`<TextField /> multiple params name + label + placeholder + errors[1] 1`] = `
<div
  className="form-item"
>
  <label
    htmlFor="TF-INPUT_NAME"
  >
    INPUT_LABEL
  </label>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    placeholder="INPUT_PLACEHOLDER"
    type="text"
  />
  <div
    className="msg msg--error"
  >
    SINGLE_ERROR
  </div>
</div>
`;

exports[`<TextField /> multiple params name + label + placeholder + errors[2] 1`] = `
<div
  className="form-item"
>
  <label
    htmlFor="TF-INPUT_NAME"
  >
    INPUT_LABEL
  </label>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    placeholder="INPUT_PLACEHOLDER"
    type="text"
  />
  <div
    className="errors-list"
  >
    <div
      className="msg msg--error"
    >
      FIRST_ERROR
    </div>
    <div
      className="msg msg--error"
    >
      SECOND_ERROR
    </div>
  </div>
</div>
`;

exports[`<TextField /> multiple params name + label + placeholder 1`] = `
<div
  className="form-item"
>
  <label
    htmlFor="TF-INPUT_NAME"
  >
    INPUT_LABEL
  </label>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    placeholder="INPUT_PLACEHOLDER"
    type="text"
  />
</div>
`;

exports[`<TextField /> multiple params name + label 1`] = `
<div
  className="form-item"
>
  <label
    htmlFor="TF-INPUT_NAME"
  >
    INPUT_LABEL
  </label>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    type="text"
  />
</div>
`;

exports[`<TextField /> multiple params name + placeholder + errors[1] 1`] = `
<div
  className="form-item"
>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    placeholder="INPUT_PLACEHOLDER"
    type="text"
  />
  <div
    className="msg msg--error"
  >
    SINGLE_ERROR
  </div>
</div>
`;

exports[`<TextField /> multiple params name + placeholder + errors[2] 1`] = `
<div
  className="form-item"
>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    placeholder="INPUT_PLACEHOLDER"
    type="text"
  />
  <div
    className="errors-list"
  >
    <div
      className="msg msg--error"
    >
      FIRST_ERROR
    </div>
    <div
      className="msg msg--error"
    >
      SECOND_ERROR
    </div>
  </div>
</div>
`;

exports[`<TextField /> multiple params name + placeholder 1`] = `
<div
  className="form-item"
>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    placeholder="INPUT_PLACEHOLDER"
    type="text"
  />
</div>
`;

exports[`<TextField /> param name: should render only wrapped input field with defaults 1`] = `
<div
  className="form-item"
>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    type="text"
  />
</div>
`;
